eng/targetingpacks.targets

```diff
@@ -0,0 +1,106 @@
+    <When Condition="'$(DisableImplicitAssemblyReferences)' == 'true' and
+                     '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                     $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '3.0'))">
+    <PropertyGroup>
+      <MicrosoftNetCoreAppRefPackRefDir>$(PkgMicrosoft_NETCore_App)\ref\$(_ShortFrameworkIdentifier)$(_ShortFrameworkVersion)\</MicrosoftNetCoreAppRefPackRefDir>
+    </PropertyGroup>
+    <ItemGroup>
+      <PackageReference Update="Microsoft.NETCore.App"
+                        ExcludeAssets="all"
+                        GeneratePathProperty="true" />
+    </ItemGroup>
+    </When>
+  </Choose>
+
+  <!-- Simple name references will be resolved from the targeting pack folders and should never be copied to the output. -->
+  <ItemGroup>
+    <Reference Update="@(Reference)">
+      <Private Condition="'%(Reference.Extension)' != '.dll'">false</Private>
+    </Reference>
+  </ItemGroup>
+
+  <!-- .NETCoreApp > 2.x DisableImplicitAssemblyReferences support. -->
+  <Target Name="RemoveFrameworkReferences"
+          BeforeTargets="_HandlePackageFileConflicts"
+          AfterTargets="ResolveTargetingPackAssets"
+          Condition="'$(DisableImplicitAssemblyReferences)' == 'true' and
+                     '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                     $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '3.0'))">
+    <PropertyGroup>
+      <MicrosoftNetCoreAppRefPackRefDir>%(ResolvedFrameworkReference.TargetingPackPath)\ref\$(_ShortFrameworkIdentifier)$(_ShortFrameworkVersion)\</MicrosoftNetCoreAppRefPackRefDir>
+    </PropertyGroup>
+    <ItemGroup>
+      <Reference Remove="@(Reference)"
+                 Condition="'%(Reference.FrameworkReferenceName)' == '$(SharedFrameworkName)'" />
+    </ItemGroup>
+  </Target>
+
+  <!-- Add the resolved targeting pack to the assembly search path. -->
+  <Target Name="UseTargetingPackForAssemblySearchPaths"
+          BeforeTargets="ResolveAssemblyReferences;
+                         DesignTimeResolveAssemblyReferences"
+          Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
+    <PropertyGroup>
+      <AssemblySearchPaths>$(AssemblySearchPaths);$(MicrosoftNetCoreAppRefPackRefDir.TrimEnd('/\'))</AssemblySearchPaths>
+      <DesignTimeAssemblySearchPaths>$(DesignTimeAssemblySearchPaths);$(MicrosoftNetCoreAppRefPackRefDir.TrimEnd('/\'))</DesignTimeAssemblySearchPaths>
+    </PropertyGroup>
+  </Target>
+
+  <!-- Use local targeting pack for NetCoreAppCurrent. -->
+  <Target Name="UpdateTargetingAndRuntimePack"
+          Condition="'$(_UseLocalTargetingRuntimePack)' == 'true'"
+          AfterTargets="ResolveFrameworkReferences">
+    <ItemGroup>
+      <ResolvedTargetingPack Path="$(MicrosoftNetCoreAppRefPackDir.TrimEnd('/\'))"
+                             NuGetPackageVersion="$(ProductVersion)"
+                             PackageDirectory="$(MicrosoftNetCoreAppRefPackDir.TrimEnd('/\'))"
```

Comment:
Why do we need to trimEnd here? Why not just TrimEnd in the property declaration instead to avoid all this TrimEnd duplication?

Reply:
For consistency with our other properties that point to a directory. Most of the time we use the NormalizeDirectory function and if not then just because of historical reasons. You would expect that the property ends with a slash. In these cases I'm trimming the trailing slash as that's the format the sdk expects. It would probably not cause any issues with the slash but I would need to look away :D